File: 0952-word-subsets/0952-word-subsets.java
class Solution {
    public List<String> wordSubsets(String[] words1, String[] words2) {
        int[] maxfreq = new int[26];
        int[] tempfreq = new int[26];
        for(String word:words2){
            Arrays.fill(tempfreq,0);
            for(char ch:word.toCharArray()){
                tempfreq[ch-'a']++;
            }
            for(int i=0;i<26;i++){
                maxfreq[i] = Math.max(maxfreq[i],tempfreq[i]);
            }
        }

        List<String> l = new ArrayList<>();
        for(String word:words1){
            Arrays.fill(tempfreq,0);
            for(char ch:word.toCharArray()){
                tempfreq[ch-'a']++;
            }
            boolean is_universal = true;
            for(int i=0;i<26;i++){
                if(maxfreq[i]>tempfreq[i]){
                    is_universal = false;
                    break;
                }
            }
            if(is_universal==true){
                l.add(word);
            }
        }
        return l;
    }
}

File: 0952-word-subsets/0952-word-subsets.py
class Solution:
    def wordSubsets(self, words1: List[str], words2: List[str]) -> List[str]:
        dict2 = Counter()
        for word in words2:
            freq = Counter(word)
            for char in freq:
                dict2[char] = max(dict2[char],freq[char])

        ans = []
        for word in words1:
            freq = Counter(word)
            is_universal = True
            for char in dict2:
                if freq[char] < dict2[char]:
                    is_universal = False
                    break
            if is_universal :
                ans.append(word)
        return ans